File: 02_io_bound/Python/main.py
import time

start = time.time()

line_count = 0
with open("../bigfile.txt", "r", encoding="utf-8") as file:
    for _ in file:
        line_count += 1

end = time.time()

print(f"Total lines: {line_count}")
print(f"Execution time: {(end - start) * 1000:.2f} ms")

File: 02_io_bound/C/main.c
#include <stdio.h>
#include <time.h>

int main() {
    FILE *file = fopen("../bigfile.txt", "r");
    if (file == NULL) {
        perror("Error opening file");
        return 1;
    }

    clock_t start = clock();

    int lineCount = 0;
    char buffer[1024];

    while (fgets(buffer, sizeof(buffer), file)) {
        lineCount++;
    }

    clock_t end = clock();
    fclose(file);

    double timeTaken = ((double)(end - start) / CLOCKS_PER_SEC) * 1000;

    printf("Total lines: %d\n", lineCount);
    printf("Execution time: %.2f ms\n", timeTaken);

    return 0;
}
